tests/integration/backup/test_v3_backup.py

```diff
@@ -1043,3 +1046,125 @@ def test_gives_non_successful_exit_code_for_legacy_backup_format(
         assert cp.returncode == 1
         assert cp.stderr.decode() == error_message
         assert cp.stdout == b""
+
+
+def test_backup_creation_fails_with_consumer_subscribe_assign(
```

Comment:
While this looks useful to pin-point the issue, as this is not a feature, I don't think it makes sense to check this test in.

What I meant to say with the need for a regression test is rather we should have a test that fails when the fix is not in place, and that passes when the fix is there. However, that should just be checked manually and we should not keep around a test to check that we're failing when the code is patched back into the bad version.

Reply:
Fair enough. Although my thinking was more along the lines of documenting behaviour with executable proof, rather than testing a feature. Nevertheless, I'll remove the 2 extra tests. 👍